dashboard/app/controllers/admin_search_controller.rb

```diff
@@ -80,18 +80,23 @@ def show_pilot
     @emails = User.where(id: user_ids).pluck(:email)
   end
 
+  # Parses newline separated emails, ignores commas and whitespace
   def add_to_pilot
-    email = params[:email]
+    emails = params[:email]
     pilot_name = params[:pilot_name]
-    return head :bad_request unless Experiment::PILOT_EXPERIMENTS.find {|p| p[:name] == pilot_name}
-    user = User.find_by_email_or_hashed_email(email)
-    if !user
-      flash[:alert] = "An account with the email address #{email} does not exist"
-    elsif user.student?
-      flash[:alert] = "Cannot add a student to the pilot"
-    else
-      SingleUserExperiment.find_or_create_by!(min_user_id: user.id, name: pilot_name)
-      flash[:notice] = "Successfully added #{email} to #{pilot_name}!"
+    email_array = emails.split("\n").map(&:strip)
```

Comment:
Looks like you have a `map` here and then an `each` below, could you combine the two (do the mapping work in the each block instead?)

Reply:
The input I'm working with is a single string, so this makes the splitting of that string explicit, allowing each email address to be handled on its own. Is there a simple way to combine them?

Reply:
So you can leave `email_array = emails.split("\n")` and then in the `each` block that first line can become `email = email.strip.gsub(/[\s,]/, "")`

Reply:
ohhhh I see what you mean. Thanks!